services/gates_services.go

```diff
@@ -0,0 +1,29 @@
+package services
+
+import (
+	"airport-app-backend/models"
+	"fmt"
+
+	"github.com/rs/zerolog/log"
+)
+
+var DEFAULT_PAGE_SIZE = 10
+
+type IGateRepository interface {
+	GetGates(int, int) ([]models.Gate, error)
+}
+
+func (sr *ServiceRepository) GetGates(page, floor int) ([]models.Gate, error) {
+	log.Debug().Msg("Fetching list of gates")
+	var gates []models.Gate
+    offset := (page - 1) * DEFAULT_PAGE_SIZE
+    query := sr.db.Offset(offset).Limit(DEFAULT_PAGE_SIZE)
+	fmt.Printf("offset : %v page : %v floor: %v", offset,page,floor)
+    if floor != -1 {
```

Comment:
So this endpoint will only return list of gates for specific floor?

Reply:
if floor is given..yes else all the gates